lib/Base/Content/Post.vala

```diff
@@ -93,6 +93,17 @@ public abstract class Backend.Post : Object {
    */
   public string? replied_to_id { get; construct; }
 
+  /**
+   * Whether the post has been favourited by the session user
+   */
+  public bool is_favourited { get; construct; default = false; }
+
+
+  /**
+   * Whether the post has been reblogged by the session user
+   */
+   public bool is_reposted { get; construct; default = false; }
+
```

Comment:
The setters should be `construct set`, probably `private construct set`. This way, you can re-set them after construction.

Reply:
My idea is to make them read-only properties that get built in the constructor, like the other properties, but there's a default because I'm not loading the Twitter values in the constructor yet. Based on [this description](https://naaando.gitbooks.io/the-vala-tutorial/content/en/4-object-oriented-programming/gobject-style-construction.html) then `get; construct;` should give Read / Write-Construct-Only.

Reply:
Just as clarification: The suggestion is only about the setter, so the changed code would be:

```
public bool is_reposted { get; private construct set; default = false; }
```

Reply:
There's too many keywords in there 😄

If I'm understanding correctly, that would let us set it in the constructor and set it at any point in the future from the class. But child classes won't be able to set it. But we don't (currently) need to do that, because we set it at construct time, or we rebuild a new object from JSON. I'll change it if the Twitter implementation needs it, but otherwise I think restrictive is better and we don't need private setter access.